src/dev-server.ts

```diff
@@ -89,14 +115,21 @@ class Server {
 
       const found = (
         getFile(url) ??
+        getFile(url + '.html') ??
         getFile(path.posix.join(url, 'index.html'))
       );
 
       if (found) {
         res.statusCode = 200;
         const contentType = mimetypes.contentType(path.extname(found.url));
         res.setHeader('content-type', contentType || 'application/octet-stream');
-        res.end(found.blob);
+        if (process.env.NODE_ENV !== 'production' && contentType.startsWith('text/html')) {
```

Comment:
Is NODE_ENV ever production? This is a dev server only. Usually it just generates static files for prod.